cap/modules/schemas/models/templates.py

```diff
@@ -0,0 +1,99 @@
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+"""Models for schema templates."""
+
+from datetime import datetime
+
+from invenio_db import db
+from invenio_search import current_search_client as es
+from sqlalchemy import event
+from sqlalchemy.orm import validates
+
+from cap.types import json_type
+
+from cap.modules.schemas.serializers import schema_template_serializer
+from cap.modules.schemas.utils import create_index
+
+
+class Template(db.Model):
+    """Model defining analysis ES template mappings."""
+
+    id = db.Column(db.Integer, primary_key=True)
+    prefix = db.Column(db.String(128), unique=False, nullable=False)
+    name = db.Column(db.String(128), unique=False, nullable=False)
+    mapping = db.Column(json_type, default=lambda: dict(), nullable=True)
+
+    created = db.Column(db.DateTime(), default=datetime.utcnow, nullable=False)
+    updated = db.Column(db.DateTime(), default=datetime.utcnow, nullable=False)
+
+    __tablename__ = 'schemas_template'
+
+    def __init__(self, *args, **kwargs):
+        """Possible to set version from string."""
+        version = kwargs.pop('version', None)
+        if version:
+            self.version = version
+
+        super().__init__(*args, **kwargs)
+
+    def __str__(self):
+        """Stringify template object."""
+        return f'{self.prefix}-{self.name}-v{self.version}'
+
+    def serialize(self):
+        """Serialize schema model."""
+        return schema_template_serializer.dump(self).data
+
+    @validates('name')
+    def validate_name(self, key, name):
+        """Validate if name is ES compatible."""
+        if any(x in r' ,"\<*>|?' for x in name):
+            raise ValueError('Name cannot contain the following characters'
+                             '[, ", *, \\, <, | , , , > , ?]')
+        return name
+
+    def update(self, **kwargs):
+        """Update schema instance."""
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+        return self
+
+
+@event.listens_for(Template, 'after_insert')
+def after_insert_schema(target, value, schema):
+    """On schema insert, create corresponding indexes and aliases in ES."""
+    create_index(schema.es_index, schema.es_mapping, schema.es_aliases)
```

Comment:
So in all of those signals this is really not what you want to do. This is just copypasted from schemas model, and it works there, because there we are creating indices for our schemas. In your case you'd like to create ES templates (which are different objects than indices). 
You can read about it [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html). As you can read there, they also have mappings field (maybe thats what was confusing), or settings, but they also have field called `index_patterns`. That's an important one, that tells ES to which indices this template should be used.